lib/new_inject.rb

```diff
@@ -0,0 +1,44 @@
+class Array
+
+  def new_inject *arg, &block
+    if arg == []
+      copy = self.dup
+      result = copy.delete_at(0)
+      copy.each do |element|
+        result = block.call(result, element)
+      end
+      result
+    else
+      result = arg[0]
+      self.each do |element|
+        result = block.call(result, element)
+      end
+      result
+    end
+  end
+
+  def recursion_inject *arg, &block
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for recursion_inject is too high. [15.56/15]